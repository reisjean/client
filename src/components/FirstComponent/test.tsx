import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the head', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /XPTO text/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': 'var(--backgroundColor)'
    })
  })
})
