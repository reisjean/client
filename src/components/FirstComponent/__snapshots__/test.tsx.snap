// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstComponent /> should render the head 1`] = `
.c0 {
  background-color: var(--bgDefault);
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c2 {
  font-size: 2.5rem;
  color: var(--primary);
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
  color: var(--secondary);
}

<main
  class="c0"
>
  <img
    alt="one day this will be a real logo, i trust this!"
    class="c1"
    src="/img/logo.png"
  />
  <h1
    class="c2"
  >
    XPTO text
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS, Styled Components & Jest
  </h2>
</main>
`;
